drivers/SmartThings/matter-sensor/src/test/test_matter_sensor_featuremap.lua

```diff
@@ -0,0 +1,180 @@
+      clusters = {
+        {cluster_id = clusters.RelativeHumidityMeasurement.ID, cluster_type = "SERVER"},
+      }
+    }
+  }
+})
+
+local mock_device_temp_humidity = test.mock_device.build_test_matter_device({
+  profile = t_utils.get_profile_definition("temperature-humidity.yml"),
+  manufacturer_info = {
+    vendor_id = 0x0000,
+    product_id = 0x0000,
+  },
+  endpoints = {
+    {
+      endpoint_id = 0,
+      clusters = {
+        {cluster_id = clusters.Basic.ID, cluster_type = "SERVER"},
+      },
+      device_types = {
+        device_type_id = 0x0016, device_type_revision = 1, -- RootNode
+      }
+    },
+    {
+      endpoint_id = 1,
+      clusters = {
+        {cluster_id = clusters.RelativeHumidityMeasurement.ID, cluster_type = "SERVER"},
+      }
+    },
+    {
+      endpoint_id = 2,
+      clusters = {
+        {cluster_id = clusters.TemperatureMeasurement.ID, cluster_type = "BOTH"},
+      }
+    }
+  }
+})
+
+local cluster_subscribe_list_humidity_battery = {
+  clusters.RelativeHumidityMeasurement.attributes.MeasuredValue,
+  clusters.PowerSource.attributes.BatPercentRemaining
+}
+
+local cluster_subscribe_list_humidity_no_battery = {
+  clusters.RelativeHumidityMeasurement.attributes.MeasuredValue
+}
+
+local cluster_subscribe_list_temp_humidity = {
+  clusters.RelativeHumidityMeasurement.attributes.MeasuredValue,
+  clusters.TemperatureMeasurement.attributes.MeasuredValue
+}
+
+local function test_init()
+  local subscribe_request_humidity_battery = cluster_subscribe_list_humidity_battery[1]:subscribe(mock_device_humidity_battery)
+  for i, cluster in ipairs(cluster_subscribe_list_humidity_battery) do
+    if i > 1 then
+      subscribe_request_humidity_battery:merge(cluster:subscribe(mock_device_humidity_battery))
+    end
+  end
+
+  local subscribe_request_humidity_no_battery = cluster_subscribe_list_humidity_no_battery[1]:subscribe(mock_device_humidity_no_battery)
+  for i, cluster in ipairs(cluster_subscribe_list_humidity_no_battery) do
+    if i > 1 then
+      subscribe_request_humidity_no_battery:merge(cluster:subscribe(mock_device_humidity_no_battery))
+    end
+  end
+
+  local subscribe_request_temp_humidity = cluster_subscribe_list_temp_humidity[1]:subscribe(mock_device_temp_humidity)
+  for i, cluster in ipairs(cluster_subscribe_list_temp_humidity) do
+    if i > 1 then
+      subscribe_request_temp_humidity:merge(cluster:subscribe(mock_device_temp_humidity))
+    end
+  end
+
+  test.socket.matter:__expect_send({mock_device_humidity_battery.id, subscribe_request_humidity_battery})
+  test.socket.matter:__expect_send({mock_device_humidity_no_battery.id, subscribe_request_humidity_no_battery})
+  test.socket.matter:__expect_send({mock_device_temp_humidity.id, subscribe_request_temp_humidity})
+  test.mock_device.add_test_device(mock_device_humidity_battery)
+  test.mock_device.add_test_device(mock_device_humidity_no_battery)
+  test.mock_device.add_test_device(mock_device_temp_humidity)
```

Comment:
maybe stagger these adds and `expect_send`s?

Reply:
Fixed by separating each device into its own `test_init` function